apps/designer/app/designer/shared/design-tokens-manager/schema.ts

```diff
@@ -1,34 +1,29 @@
 import { z } from "zod";
 import { groups } from "./groups";
 
-type UnionToIntersection<U> = (
-  U extends never ? never : (arg: U) => never
-) extends (arg: infer I) => void
-  ? I
-  : never;
+const mapToProp = <
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends ReadonlyArray<any>,
+  Prop extends keyof T[number]
+>(
+  arr: readonly [...T],
+  prop: Prop
+): {
+  [K in keyof T]: T[K][Prop];
+} => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return arr.map((item) => item[prop]) as any;
+};
 
-type UnionToTuple<T> = UnionToIntersection<
-  T extends never ? never : (t: T) => T
-> extends (_: never) => infer W
-  ? [...UnionToTuple<Exclude<T, W>>, W]
-  : [];
-
-type Type = typeof groups[number]["type"];
-type Group = typeof groups[number]["group"];
-
-// zod is super restrictive on enum type https://github.com/colinhacks/zod/issues/1118
-// so we convert union to tuple here its seriously safer than any as it will break on things like
-// ['a', 'b'] as UnionToTuple<Type> etc
-const types = groups.map((group) => group.type) as UnionToTuple<Type>;
-const groupNames = groups.map((group) => group.group) as UnionToTuple<Group>;
+// zod expects readonly tuples for its enums
+const types = mapToProp(groups, "type");
+const groupNames = mapToProp(groups, "group");
 
 export const DesignToken = z.object({
   name: z.string(),
-  // @todo if https://github.com/colinhacks/zod/issues/1118
```

Comment:
I think that this issue was linked here incorrectly. The issue is about the fact that `enum` requires strings - but we have strings here so it wasn't really applicable to this issue here.